### DISCUSSION TITLE
Sampling vs where filters

### DISCUSSION THREAD

**Discussion by adam-mrozik** [ClickHouse/ClickHouse#59995] [Q&A]
Hey guys,

I have a simple question.
If I write a query like this:
```
SELECT toStartOfHour(timestamp) as timestamp, count(*)
FROM table SAMPLE 1000
WHERE country = "US"
GROUP BY timestamp
```
(partitioning key is timestamp daily)
Will where filters be applied before sampling?

From what I understood, SAMPLE 1000 will take at least 1000 events, and I assume it will take roughly same amount from events for each day.

I assume sensible thing would be to first filter by country = US, before applying sampling; otherwise it wouldn't be representative, as sample could take, let's say, 1000 events from other countries and present me with a false correlation between daily counts. 

Could you confirm my understanding of how sampling works is correct?

**Comment by Twixes** [comment]
I'm not a ClickHouse dev, but it's really important what's the schema, and particularly what the sampling key is. If it's something unrelated to `country`, then there should be no false correlation.

1000 rows is an extremely small number for ClickHouse though (the default granule size is 8192 rows), so the quality of sampling will be poor anyway if this is a real number you're using. It should be tens of thousands minimum in the `SAMPLE` clause.

**Comment by geranazavr555** [comment]
I have the same question and found no information about it - does clickhouse applying sample before filters (in the where-clause) or after that
